src/plots/barPlot.ts

```diff
@@ -751,7 +752,11 @@ export class Bar<X, Y> extends XYPlot<X, Y> {
     const thicknessF = attrToProjector[Bar._BAR_THICKNESS_KEY];
     const gapF = attrToProjector["gap"];
     const thicknessMinusGap = gapF == null ? thicknessF : (d: any, i: number, dataset: Dataset) => {
-      return thicknessF(d, i, dataset) - gapF(d, i, dataset);
+      const thick = thicknessF(d, i, dataset);
+      // only subtract gap if bars are at least 2 pixels wide, otherwise canvas
+      // interpolation can cause bars to become invisible due to subpixel
+      // sampling
```

Comment:
tiny nit: put this word on the previous line. shorter file!

Reply:
i autowrap comments to 80 chars for readability